src/common/arrstr.cpp

```diff
@@ -573,3 +579,169 @@ wxArrayString wxSplit(const wxString& str, const wxChar sep, const wxChar escape
+            return fragment;
+
+        if (naturalNumeric.Matches(text))
+        {
+            naturalNumeric.GetMatch(&digitStart, &digitLength, 0);
+        }
+
+        if (naturalAlpha.Matches(text))
+        {
+            naturalAlpha.GetMatch(&alphaStart, &alphaLength, 0);
+        }
+
+
+        if (alphaStart == 0)
+        {
+            fragment.text = text.Mid(0, alphaLength);
+            fragment.value = 0;
+            fragment.type = wxFRAGMENT_TYPE_ALPHA;
+
+            text.erase(0, alphaLength);
+        }
+
+        if (digitStart == 0)
+        {
+            fragment.text = text.Mid(0, digitLength);
+            fragment.text.ToDouble(&fragment.value);
+            fragment.type = wxFRAGMENT_TYPE_DIGIT;
+
+            text.erase(0, digitLength);
+        }
+
+        return fragment;
+    }
+
+    int CompareFragmentNatural(const wxStringFragment &lhs, const wxStringFragment &rhs, bool noCase)
+    {
+        if ((lhs.type == wxFRAGMENT_TYPE_ALPHA) &&
+            (rhs.type == wxFRAGMENT_TYPE_ALPHA))
+        {
+            if (noCase)
+                return lhs.text.CmpNoCase(rhs.text);
+            else
+                return lhs.text.Cmp(rhs.text);
+        }
+
+        if ((lhs.type == wxFRAGMENT_TYPE_DIGIT) &&
+            (rhs.type == wxFRAGMENT_TYPE_DIGIT))
+        {
+            if (lhs.value == rhs.value)
+            {
+                return  0;
+            }
+
+            if (lhs.value < rhs.value)
+            {
+                return -1;
+            }
+
+            if (lhs.value > rhs.value)
+            {
+                return  1;
+            }
+        }
+
+        if ((lhs.type == wxFRAGMENT_TYPE_DIGIT) &&
+            (rhs.type == wxFRAGMENT_TYPE_ALPHA))
+        {
+            return -1;
+        }
+
+        if ((lhs.type == wxFRAGMENT_TYPE_ALPHA) &&
+            (rhs.type == wxFRAGMENT_TYPE_DIGIT))
+        {
+            return 1;
+        }
+
+        if (lhs.type == wxFRAGMENT_TYPE_EMPTY)
+        {
+            return -1;
+        }
+
+        if (rhs.type == wxFRAGMENT_TYPE_EMPTY)
+        {
+            return 1;
+        }
+
+        return 0;
+    }
+
+
+    inline int CompareNaturalFunction(const wxString &s1, const wxString &s2, bool noCase)
+    {
+        wxString lhs(s1);
+        wxString rhs(s2);
+        wxStringFragment fragL;
+        wxStringFragment fragR;
+
+        int comparison = 0;
+
+        while ((comparison == 0) && (lhs.Length() || rhs.Length()))
+        {
+            fragL = GetFragment(lhs);
+            fragR = GetFragment(rhs);
+            comparison = CompareFragmentNatural(fragL, fragR, noCase);
+        }
+
+        return comparison;
+    }
+
+} // unnamed namespace
+
+
+WXDLLIMPEXP_BASE
+int wxCMPFUNC_CONV wxCmpNatural(const wxString& s1, const wxString& s2)
+{
+    #ifdef __WXMSW__
+        return StrCmpLogicalW(s1.wc_str(), s2.wc_str());
+    #else
+        bool noCase = true;
+        return CompareNaturalFunction(s1, s2, noCase);
```

Comment:
Is there a reason why a local variable is created here, instead of just calling `return CompareNaturalFunction(s1, s2, true);`?

Reply:
Yes, it's to make it clear the meaning of that argument, without the reader having to go and find the definition. Simply having CompareNaturalFunction(s1, s2, true) would be a little opaque.

Reply:
IMO: If you feel that the meaning of the parameter here is really important, that's what comments are for. I find creating a local variable instead a bit unusual...

Reply:
Now that I think about it, why don't I just get rid of all the case / nocase options. Is anyone likely to want natural sort with case sensitivity ?